osconfig_tests/test_suites/package_management/package_management_utils.go

```diff
@@ -159,7 +160,53 @@ func getPackageInstallRemovalStartupScript(pkgManager, packageName string) strin
 		ss = fmt.Sprintf(ss, utils.InstallOSConfigYumEL7, packageName, packageInstalledString, packageNotInstalledString)
 
 	default:
-		panic(fmt.Sprintf("invalid package manager: %s", pkgManager))
+		logger.Errorf(fmt.Sprintf("invalid package manager: %s", pkgManager))
+	}
+
+	return ss
+}
+
+func getPackageInstallFromNewRepoTestStartupScript(pkgManager, packageName string) string {
+	var ss string
+
+	switch pkgManager {
+
+	case "apt":
+		ss = "%s\n" +
+			"sleep 10;\n" +
```

Comment:
Why sleep for 10 seconds? Is it trying to wait for something to happen that could, in some circumstances, take longer than 10 seconds?

Reply:
while testing it manually,  i noticed that the agent needed a restart. Did not have to wait before restart. to be on the safe side, i added the wait.
I am actually curious to know why the agent needs a restart, ideally it should have worked on the first run itself.
I am thinking of adding debug flag through instancemetadata so the agent can run in debug mode.

Reply:
Sounds good. Can you add a comment to the code explaining that? @adjackura is this related to agents hanging on startup? If so, the fact that a restart fixes the problem may be a new clue.

Reply:
added a comment explaining the test runner behavior